examples/visualize_features.py

```diff
@@ -125,11 +142,17 @@ def main(rank: int = 0, world_size: int = 1):
                                        pad_mean=preprocessor.norm_mean if isinstance(preprocessor, InputConditioner) else None,)
 
     if not os.path.isdir(args.dataset):
-        ds_builder = load_dataset_builder(args.dataset, trust_remote_code=True)
+        if args.data_dir is not None:
+            ds_builder = load_dataset_builder("imagefolder", data_dir=args.data_dir)
+        else:
+            ds_builder = load_dataset_builder(args.dataset, trust_remote_code=True)
         dataset = ds_builder.as_dataset(split=args.split)
         dataset = dataset.to_iterable_dataset(num_shards=world_size * max(1, args.workers))
         dataset = split_dataset_by_node(dataset, rank=rank, world_size=world_size)
-        dataset = dataset.map(lambda ex: dict(image=transform(ex['image']), label=torch.as_tensor(ex['label'], dtype=torch.int64)))
+        if args.data_dir is not None:
+            dataset = dataset.map(lambda ex: dict(image=transform(ex['image']), label=torch.as_tensor(0, dtype=torch.int64)))
+        else:
+            dataset = dataset.map(lambda ex: dict(image=transform(ex['image']), label=torch.as_tensor(ex['label'], dtype=torch.int64)))
```

Comment:
Do we even care about label? Could probably just set it to 0 in all cases.

Reply:
OK yes, I've changed the code as you suggested.